### PR TITLE
postgresql: update to 14.1

### PR DESCRIPTION
**Pull request by zorro**

### CODE REVIEW

File: packages/postgresql/fix-hardcoded-paths.patch [hunk 1]
```diff
@@ -51,15 +51,6 @@
  	write_stderr(_("%s: could not start server: %s\n"),
 --- ./src/bin/pg_upgrade/check.c	2021-05-11 02:11:42.000000000 +0530
 +++ ./src/bin/pg_upgrade/check.c.mod	2021-06-17 12:21:44.835155788 +0530
-@@ -471,7 +471,7 @@
- 
- #ifndef WIN32
- 	/* add shebang header */
--	fprintf(script, "#!/bin/sh\n\n");
-+	fprintf(script, "#!@TERMUX_PREFIX@/bin/sh\n\n");
```
Comment (thunder-coding) [inline packages/postgresql/fix-hardcoded-paths.patch:59]: Why is this being removed? The patch is still needed

**Review by thunder-coding — CHANGES_REQUESTED** [review]

File: packages/postgresql/fix-hardcoded-paths.patch (hunk 1 above)
Comment (xeffyr) [reply packages/postgresql/fix-hardcoded-paths.patch:59]: Confirming, this line is still present in git and should be fixed by patch: https://github.com/postgres/postgres/blob/0a455b8d61d8fc5a7d1fdc152667f9ba1fd27fda/src/bin/pg_upgrade/check.c#L580

All Termux packages should have patches for hardcoded paths. It doesn't matter whether `termux-exec` can handle standard shebangs. That thing is for users stuff, not saying that it still doesn't work on some devices.

Comment (zorro) [reply packages/postgresql/fix-hardcoded-paths.patch:59]: @xeffyr There used to be two hardcoded paths in that file. Now there's only one. One got removed on version 14. Here the file from version 13:

https://github.com/postgres/postgres/blob/272d82ec6febb97ab25fd7c67e9c84f4660b16ac/src/bin/pg_upgrade/check.c#L472

**Review by Grimler91 — APPROVED** [review]

File: packages/postgresql/fix-hardcoded-paths.patch (hunk 1 above)
Comment (Grimler91) [reply packages/postgresql/fix-hardcoded-paths.patch:59]: Looks correct

**Review by thunder-coding — APPROVED** [review]

**Pull request merged**